app/views/ProjectEdit/GeoAreas/RegionCard/index.tsx

```diff
@@ -154,72 +214,97 @@ function RegionCard(props: Props) {
+        {
+            loading: pendingDeleteRegion,
         },
-    });
+    ] = useMutation<DeleteRegionMutation, DeleteRegionMutationVariables>(
+        DELETE_REGION,
+        {
+            onCompleted: (response) => {
+                if (!response.project?.projectRegionBulk) {
+                    return;
+                }
+
+                const {
+                    deletedResult,
+                    errors,
+                } = response.project.projectRegionBulk;
+
+                if (errors) {
+                    alert.show(
+                        'Failed to publish selected region.',
+                        { variant: 'error' },
+                    );
+                }
+
+                const ok = deletedResult?.some((result) => result.id === region.id);
+
+                if (ok) {
+                    alert.show(
+                        'Region is successfully deleted!',
+                        { variant: 'success' },
+                    );
+                    onRegionRemoveSuccess();
+                }
+            },
+            onError: () => {
+                alert.show(
+                    'Failed to delete selected region.',
+                    { variant: 'error' },
+                );
+            },
+        },
+    );
 
     const handleDeleteRegionClick = useCallback(
         () => {
-            if (regions) {
-                removeRegion(null);
+            if (region.id) {
+                deleteRegion({
+                    variables: {
+                        projectId: activeProject,
+                        regionId: [region.id],
+                    },
+                });
             }
         },
-        [removeRegion, regions],
+        [deleteRegion, region, activeProject],
     );
 
     const handleAdminLevelAdd = useCallback(
         () => {
             if (!onTempAdminLevelChange) {
                 return;
             }
-            const clientId = randomString();
             const newAdminLevel: PartialAdminLevel = {
-                clientId,
                 title: `Admin Level ${adminLevelsLength}`,
                 level: adminLevelsLength,
-                region: +region.id,
             };
             onTempAdminLevelChange(newAdminLevel);
         },
-        [region.id, adminLevelsLength, onTempAdminLevelChange],
+        [adminLevelsLength, onTempAdminLevelChange],
     );
 
     const handleAdminLevelSave = useCallback(
-        (value: AdminLevelGeoArea) => {
+        (value: AdminLevelType) => {
             if (onTempAdminLevelChange) {
                 onTempAdminLevelChange(undefined);
             }
 
             setAdminLevels((oldAdminLevels) => {
                 const newAdminLevels = [...oldAdminLevels];
                 const index = newAdminLevels.findIndex((item) => item.id === value.id);
-                const valueWithClientId = {
-                    ...value,
-                    clientId: value.id.toString(),
-                };
```

Comment:
Are you sure we don't need this?

Reply:
yes i checked it, we don't need it anymore